cmd/cluster/health.go

```diff
@@ -85,101 +102,158 @@ func (o *healthOptions) run() error {
 		return err
 	}
 	cluster := clusterResp.Body()
-	healthObject := createHealthObject(cluster)
+	if !cluster.Hypershift().Enabled() {
+		healthObject := createHealthObject(cluster)
 
-	if cluster.Nodes().AutoscaleCompute().MinReplicas() != 0 {
-		min := strconv.Itoa(cluster.Nodes().AutoscaleCompute().MinReplicas())
-		max := strconv.Itoa(cluster.Nodes().AutoscaleCompute().MaxReplicas())
-		healthObject.Expected.Worker = string(fmt.Sprintf("%v - %v", min, max))
-	}
-	if cluster.Nodes().Compute() != 0 {
-		healthObject.Expected.Worker = int(cluster.Nodes().Compute())
-	}
+		if cluster.Nodes().AutoscaleCompute().MinReplicas() != 0 {
+			min := strconv.Itoa(cluster.Nodes().AutoscaleCompute().MinReplicas())
+			max := strconv.Itoa(cluster.Nodes().AutoscaleCompute().MaxReplicas())
+			healthObject.Expected.Worker = string(fmt.Sprintf("%v - %v", min, max))
+		}
+		if cluster.Nodes().Compute() != 0 {
+			healthObject.Expected.Worker = int(cluster.Nodes().Compute())
+		}
 
-	runningMasters := 0
-	runningInfra := 0
-	runningWorkers := 0
-	totalStopped := 0
-	totalCluster := 0
-
-	var clusterHealthClient osdCloud.ClusterHealthClient
-	var ownedLabel string
-	infraID := cluster.InfraID()
-	if cluster.CloudProvider().ID() == "gcp" {
-		clusterHealthClient, err = osdCloud.NewGcpCluster(ocmClient, o.clusterID)
+		runningMasters := 0
+		runningInfra := 0
+		runningWorkers := 0
+		totalStopped := 0
+		totalCluster := 0
+
+		var clusterHealthClient osdCloud.ClusterHealthClient
+		var ownedLabel string
+		infraID := cluster.InfraID()
+		if cluster.CloudProvider().ID() == "gcp" {
+			clusterHealthClient, err = osdCloud.NewGcpCluster(ocmClient, o.clusterID)
+			if err != nil {
+				return err
+			}
+			ownedLabel = "kubernetes-io-cluster-" + infraID
+			defer clusterHealthClient.Close()
+		} else if cluster.CloudProvider().ID() == "aws" {
+			clusterHealthClient, err = osdCloud.NewAwsCluster(ocmClient, o.clusterID, o.awsProfile)
+			if err != nil {
+				return err
+			}
+			ownedLabel = "kubernetes.io/cluster/" + infraID
+			defer clusterHealthClient.Close()
+		} else {
+			return errors.New(fmt.Sprintf("Unknown cloud provider found: %s", cluster.CloudProvider().ID()))
+		}
+		err = clusterHealthClient.Login()
 		if err != nil {
 			return err
 		}
-		ownedLabel = "kubernetes-io-cluster-" + infraID
-		defer clusterHealthClient.Close()
-	} else if cluster.CloudProvider().ID() == "aws" {
-		clusterHealthClient, err = osdCloud.NewAwsCluster(ocmClient, o.clusterID, o.awsProfile)
+		for _, zone := range clusterHealthClient.GetAZs() {
```

Comment:
For each of these requests that reach out (to OCM, via oc get, possibly to the cloud provider) would you be opposed to using a waitGroup (https://pkg.go.dev/sync#example-WaitGroup) to run them all async and then print the results when the last one is done?

There was great success with the cluster-context subcommand in speeding up the time it takes to gather the info.  This one is kinda on the line since there's less calls, but they're all reaching out to a remote api in one way or another, so there should still be a benefit.  It might be nice to incorporate it now so in the long run as this subcommand is expanded, folks will use the waitGroup by default.

Less time for SREs to sit an wait for output :)